src/Workspaces/Core/Portable/NamingStyles/Serialization/NamingStylePreferences.cs

```diff
@@ -16,7 +16,7 @@ namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
     /// </summary>
     internal class NamingStylePreferences : IEquatable<NamingStylePreferences>
     {
-        private const int s_serializationVersion = 4;
+        private const int s_serializationVersion = 5;
```

Comment:
what impact does thsi have on people upgrading?  will their existing preferences be honored?

Reply:
I don't think so :/ this should reset your preferences

Reply:
I'll take a look at what I can do, but this is really hard to test

Reply:
If we don't increase the version, the impact on newer versions would be that your setting for methods now apply to local functions as well.

Reply:
ok.  but you've confirmed that if you do update the version then previous settings are preserved and not wiped out?

Can we just write code that migrates across versions, and converts an existing method rule so that it only applies to ordinary methods, and not local functions?  (we def need tests here as well).

Reply:
I haven't checked or tested this at all, I'll see what I can do.

Reply:
the code in FromXElement a few lines below does wipe them out

Reply:
> the code in FromXElement a few lines below does wipe them out

Ok.  we probably need to fix that.  and have this code actually upgrade things from v4 to v5.

Reply:
note: the upgrading codpath should run all the time, and it shoudl upgrade whenever the 'prior' value is before the change point.

Reply:
i can't find any code in this PR related to ensuring user preferences are not wiped out.  can you point me to it?

Reply:
fixed